File: library-infrastructure/src/main/java/com/dodecaedro/library/infrastructure/search/BookSearchDocument.java
package com.dodecaedro.library.infrastructure.search;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document(indexName = "library", type = "book")
public class BookSearchDocument {
	@Id
	private UUID id;
	@Field(type = FieldType.Text)
	private String title;
}

File: library-infrastructure/src/main/kotlin/com/dodecaedro/library/infrastructure/search/BookSearchDocument.kt
package com.dodecaedro.library.infrastructure.search

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType
import java.util.*

@Document(indexName = "library")
data class BookSearchDocument(
	@Id
	var id: UUID?,
	@Field(type = FieldType.Text)
	val title: String?
)
